http-clients/src/main/java/com/palantir/remoting/http/QueryMap.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.remoting.http;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.Multimap;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+
+@JsonDeserialize(as = ImmutableQueryMap.class)
+@JsonSerialize(as = ImmutableQueryMap.class)
+@Value.Immutable
+@Value.Style(visibility = ImplementationVisibility.PACKAGE)
+public abstract class QueryMap {
+    public abstract Multimap<String, String> queryMap();
```

Comment:
Yes -- just tried it in a unit test, and Multimaps can be written to JSON and read back from JSON properly (at least when using the JDK 7 Guava ObjectMapper).

However, after thinking about it, this object doesn't actually need to support JSON serialization since it's converted to query parameters on the client side and is never actually passed over the wire. Will remove the annotations.